fedbiomed/common/secagg/_dh.py

```diff
@@ -0,0 +1,159 @@
+            public_key_pem: Optional. The public key in PEM format.
+        """
+        if private_key_pem:
+            try:
+                self._private_key = serialization.load_pem_private_key(
+                    private_key_pem, password=None, backend=default_backend()
+                )
+            except ValueError:
+                raise FedbiomedSecaggCrypterError(
+                    f"{ErrorNumbers.FB628.value}: Invalid private key format"
+                )
+        else:
+            self._private_key = ec.generate_private_key(
+                ec.SECP256R1(), default_backend()
+            )
+
+        self._public_key = self._private_key.public_key()
+
+        if public_key_pem:
+            try:
+                self._public_key = serialization.load_pem_public_key(
+                    public_key_pem, backend=default_backend()
+                )
+            except ValueError:
+                raise FedbiomedSecaggCrypterError(
+                    f"{ErrorNumbers.FB629.value}: Invalid public key format"
+                )
+
+    @property
+    def export_private_key(self):
+        """
+        Exports the private key to PEM format.
+
+        Returns:
+            The private key in PEM format.
+        """
+        return self._private_key.private_bytes(
+            encoding=serialization.Encoding.PEM,
+            format=serialization.PrivateFormat.PKCS8,
+            encryption_algorithm=serialization.NoEncryption(),
+        )
+
+    @property
+    def export_public_key(self):
+        """
+        Exports the public key to PEM format.
+
+        Returns:
+            The public key in PEM format.
+        """
+        return self._public_key.public_bytes(
+            encoding=serialization.Encoding.PEM,
+            format=serialization.PublicFormat.SubjectPublicKeyInfo,
+        )
+
+
+class DHKeyAgreement:
+    """
+    Key Agreement with ephemeral-ephemeral ECDH, a session is defined for each experiment.
+
+    This class handles the key agreement process using the private ECC key of the user. It imports
+    the private key, performs the key exchange, and derives the shared secret using a KDF.
+
+    Attributes:
+        private_key: The user's private ECC key.
+    """
+
+    def __init__(self, node_u_id, node_u_private_key, session_salt):
+        """
+        Initializes the DHKeyAgreement instance.
+
+        Args:
+            node_u_id: The ID of the node.
+            node_u_private_key: The private key in PEM format as bytes.
+            session_salt: A session-specific salt.
+        """
+        self._node_u_id = node_u_id
+        try:
+            self._private_key = serialization.load_pem_private_key(
+                node_u_private_key, password=None, backend=default_backend()
+            )
+        except ValueError:
+            raise FedbiomedSecaggCrypterError(
+                f"{ErrorNumbers.FB628.value}: Invalid private key format"
+            )
```

Comment:
Hi @rtaiello,

Thank you for updating `DHKey` class. I think you can use DHKey here with respect to your changes in DHKey. Because DHKey already implement key import.  

```suggestion
        self._private_key = DHKey(private_key_pem=node_u_private_key)
```